test/README.md

```diff
@@ -40,6 +40,61 @@ Deciding where to put a test is (like naming things) a hard problem:
   a lot of styles consider adding a component (that doesn't require any interaction)
   to `test/regressions/tests/` e.g. `test/regressions/tests/List/ListWithSomeStyleProp`
 
+### Unexpected calls to `console.error` or `console.war`
+
+By default our test suite fails if any test recorded `console.error` or `console.warn` calls:
+![unexpected console.error call](./unexpected-console-error-call.png)
+
+The failure message includes the name of the test.
+The logged error is prefixed with the test file so that you can find the failing test faster (test file + test name).
+The error includes the logged message as well as the stacktrace of that message.
+Unfortunately the stacktrace is currently duplicated due to `chai`.
+
+However, in watchmode (`yarn test:unit --watch`) unexpected calls are ignored (see https://github.com/mochajs/mocha/issues/4347).
+You can explicitly [expect no console calls](#writing-a-test-for-consoleerror-or-consolewarn) for when you're adding a regression test.
+This makes the test more readable and properly fails the test in watchmode if the test had unexpected `console` calls.
+
+### Writing a test for `console.error` or `console.warn`
+
+If you add a new warning via `console.error` or `console.warn` you should add tests that expect this message.
+For tests that expect a call you can use our custom `toWarnDev` or `toErrorDev` matchers.
+The expected messages must be a subset of the actual messages and match the casing.
+The order of these message must match as well.
+
+Example:
+
+```jsx
+function SomeComponent({ variant }) {
+  if (process.env.NODE_ENV !== 'production') {
+    if (variant === 'unexpected') {
+      console.error("That variant doesn't make sense.");
+    }
+    console.error('The component is deprecated');
+  }
+
+  return <div />;
+}
+expect(() => {
+  render(<SomeComponent variant="unexpected" />);
+}).toErrorDev(["That variant doesn't make sense.", 'The component is deprecated']);
+```
+
+```js
+function SomeComponent({ variant }) {
+  if (process.env.NODE_ENV !== 'production') {
+    if (variant === 'unexpected') {
+      console.error("That variant doesn't make sense.");
+    }
+    console.error('The component is deprecated');
```

Comment:
this logs if variant is valid?

Reply:
Yeah, does the message indicate otherwise? It's only here to illustrate that every warning needs to be expected. These are different warnings. They would basically live in separate `process.env.NODE_ENV` blocks. I put them into a single one for brevity.

Reply:
Should we add Something like Passes: Fails: to the examples then? It’s not clear to me we expected the second one to fail.

Reply:
Ah now I get it! It's more about the expectation.